### PR TITLE
Fixes meta category paging & category selection

### PR DESCRIPTION
**Pull request by steam**
Meta categories page correctly when visited from Discover landing screen
[fixes #134596259](https://www.pivotaltracker.com/story/show/134596259)

Categories have a selected state in the horizontal category nav
[fixes #134596301](https://www.pivotaltracker.com/story/show/134596301)

Additionally, viewing a meta category from Discover scrolls to the correct navigation tile.

### CODE REVIEW

File: Sources/Controllers/Categories/CategoryViewController.swift [hunk 1]
```diff
@@ -168,26 +174,18 @@ extension CategoryViewController: CategoryStreamDestination, StreamDestination {
...
+        allCategories = categories
 
         let shouldAnimate = !(screen.navBarsVisible ?? false)
         let info = allCategories.map { (category: Category) -> CategoryCardListView.CategoryInfo in
             return CategoryCardListView.CategoryInfo(title: category.name, imageURL: category.tileURL)
         }
         screen.setCategoriesInfo(info, animated: shouldAnimate)
 
-        let selectedCategoryIndex = allCategories.indexOf { $0.id == category?.id }
+        let selectedCategoryIndex = allCategories.indexOf { $0.slug == slug }
```
Comment (steam) [inline Sources/Controllers/Categories/CategoryViewController.swift]: This change enables scrolling and selection of the meta categories.

File: Ello.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1164,7 +1165,7 @@
 		122B190F1A8692C700247A05 /* StreamImageCell.xib */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = file.xib; path = StreamImageCell.xib; sourceTree = "<group>"; };
 		122B19121A8695BF00247A05 /* StreamTextCell.xib */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = file.xib; path = StreamTextCell.xib; sourceTree = "<group>"; };
 		122B19151A86972900247A05 /* StreamFooterCell.xib */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = file.xib; path = StreamFooterCell.xib; sourceTree = "<group>"; };
-		122B30F61C6A41CF00C1B9D8 /* RegionKindStreamCellTypeAddition.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; name = RegionKindStreamCellTypeAddition.swift; path = RegionKindStreamCellTypeAddition.swift; sourceTree = "<group>"; };
+		122B30F61C6A41CF00C1B9D8 /* RegionKindStreamCellTypeAddition.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = RegionKindStreamCellTypeAddition.swift; sourceTree = "<group>"; };
```
Comment (steam) [inline Ello.xcodeproj/project.pbxproj]: Any ideas what these are about? Are they related to the changes you made recently?

File: Sources/Views/CategoryCardListView.swift [hunk 3]
```diff
@@ -77,43 +77,38 @@ public class CategoryCardListView: UIView {
...
-            let attributedString = NSAttributedString(info.title, color: .whiteColor(), alignment: .Center)
-            button.setAttributedTitle(attributedString, forState: UIControlState.Normal)
-            view.addSubview(button)
-            button.snp_makeConstraints { $0.edges.equalTo(view).inset(5) }
-
-            buttonIndexLookup[button] = index
-            index += 1
-            return view
+        categoryViews = categoriesInfo.enumerate().map { (index, info) in
+            return categoryView(index: index, info: info)
```
Comment (colinta) [inline Sources/Views/CategoryCardListView.swift]: Very nice 👍

File: Ello.xcodeproj/project.pbxproj (hunk 2 above)
Comment (colinta) [inline Ello.xcodeproj/project.pbxproj]: Yes and no - I made the changes in Sublime, and made sure the project opened, but I'm guessing that these changes (removing the `name=` property) were a result.

**Pull request merged**